pkg/ccl/serverccl/tenant_test_utils.go

```diff
@@ -141,7 +141,7 @@ var _ TenantTestHelper = &tenantTestHelper{}
 
 // NewTestTenantHelper constructs a TenantTestHelper instance.
 func NewTestTenantHelper(
-	t *testing.T, tenantClusterSize int, knobs base.TestingKnobs,
+	t *testing.T, tenantClusterSize int, numNodes int, knobs base.TestingKnobs,
```

Comment:
This API is sus...
(it was already somewhat sus before your change)

This parameter "tenantClusterSize" indicates the number of tenant servers to start _connecting to a single KV node_.  

On the one hand I get that the original goal was simply to obtain a test framework where there were multiple SQL pods running side-by-side. 
On the other hand, this is not a realistic deployment scenario -- in practice a cluster should have multiple nodes. Multiple tenant servers on top of a one-node CockroachDB cluster is not very representative.

What should we do about this?

I'm not exactly sure, but the API extension you're proposing here is adding a problem on top of another problem: now with your change, if someone specifies "tenantClusterSize=3" and "numNodes=3" they get 9 SQL servers, and I don't think that's what we want.

If I were you I would reason about this backwards: as a CRL developer, what am I trying to achieve with my test code?

It seems to me (if my reading is right) that we want 
1) some tests in this package want multiple SQL servers
2) some tests need multiple KV nodes
3) there's no test that really _requires_ to have just 1 KV node or 1 SQL server

So why not... removing these 2 parameters altogether, and always start a 3-node storage cluster with 1 tenant server per node?

Reply:
@herkolategan can I get your input on this too? Thanks